reports/gitlab-commit-protocol/gitlab-commit-protocol.tex

```diff
@@ -159,8 +159,29 @@ \section{Distributed fetches}
 TODO: To unite this section with Mutator RPC handling strategy.
 
 \section{Termination protocol}
+\subsection{Transaction timeout}
+In \emph{\href{https://gitlab.com/gitlab-org/gitaly/-/tags/v14.0.0-rc1}{ current}} 
+Gitaly version there is 5 minutes transaction timeout that is set on transaction 
+context when voting client on gitaly node votes on transaction (the voting client 
+communicates with Praefect node).
+The timeout used for all transactional actions like voting and stopping of transactions.
+ The reasons of setting of this long timeout you can find 
+In \emph{\href{https://gitlab.com/gitlab-org/gitaly/-/blob/master/internal/gitaly/transaction/manager.go\#L21}{ source code comments}}. 
+In other words It shouldn't normally be hit, but if it is hit then it indicates a real problem. 
```

Comment:
What is "a real problem"? Could you describe it please. What are possible scenarios if the connection is closed by timeout?

Reply:
@g4s8 https://github.com/cqfn/degitx/pull/203/commits/428bc3e443b5f05c446e77be5b8876f45af91eea tried to describe in this commit.